### ISSUE TITLE
Data frame case sensitivity is not carried over after Merge Transform is applied

### ISSUE THREAD

**Issue by yakshpatell** [data-forge/data-forge-ts#145]
This is a very similar issue to an earlier resolved [issue](https://github.com/data-forge/data-forge-ts/issues/138) but this time, with a different transform: MERGE.

The merge transform does not support case sensitivity. We have multiple customers requesting this feature in our application so if you could add this support, I would greatly appreciate it.

Example:

```
const frame1 = new DataFrame({
columnNames: ['Name'],
caseSensitive: true
});

const frame2 = new DataFrame({
columnNames: ['name'],
caseSensitive: true
});

const frame3 = frame1.merge(frame2);
console.log('column names: ' + JSON.stringify(frame3.getColumnNames()))

/* ^ logging returns ["Name.1","name.2"] */
```

**Comment by ashleydavis** [comment]
Thanks for your feedback.

This issue happens because the `DataFrame` that is created in the `merge` function is case insensitive by default:

![image](https://user-images.githubusercontent.com/625126/175199765-819cd2f8-2c2b-4ee1-8015-052790bdf7c9.png)

We could make it case sensitive by just adding that field:

![image](https://user-images.githubusercontent.com/625126/175199813-9da825e6-290d-4659-ae15-14a86b7c3096.png)

However, there are cases where this might not be desired.

I would suggest that we make the result case sensitive if any of the input data frames is case sensitive, like this:

![image](https://user-images.githubusercontent.com/625126/175200658-0333296a-1e25-4747-a7df-2618a208f3b5.png)

This would solve your problem and hopefully not cause an issue for anyone relying on it already.

**Comment by yakshpatell** [comment]
Ahh ok. This makes sense. Thank you for the neat step by step explanation!! Really appreciate it.

When can I expect this code enhancement to be pushed ? :))

**Comment by ashleydavis** [comment]
This issue is fixed in v1.9.6.

The `merge` function now returns a case sensitive dataframe when any of the input dataframes is case sensitive.

Thanks for reporting the issue.
